Web/Scotty/Internal/Types.hs

```diff
@@ -23,6 +23,9 @@ import           Control.Monad.Trans.Except
 import qualified Data.ByteString as BS
 import           Data.ByteString.Lazy.Char8 (ByteString)
 import           Data.Default.Class (Default, def)
+#if MIN_VERSION_base(4,9,0)
+import Data.Semigroup as Sem
```

Comment:
Perhaps we should use `Prelude.Compat` from `base-compat-batteries` to ensure that `Semigroup(..)`/`Monoid(..)` are always in scope? That way we can avoid having to muck about with CPP here.

This suggestion isn't strictly related to the PR at hand, but it might make the backwards compatibility easier to manage in the long run.